### PR TITLE
Update buildtools and CoreFx pkg dependency versions.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* Buildtools and Microsoft.Private.PackageBaseline (in CoreFx) have been updated to support 'PreReleaseGroup'.
* This means that now in VSTS, the test runs will honor the package versions we specified for non-WCF packages.
* The test runs will only ignore the specified WCF package versions in order to use the "just built" versions.
* Also pulled in necessary changes due to dotnet/corefx#11387
* Updated CoreFx, CoreClr and External package versions.

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
@dagood and @ericstj

File: pkg/baseline/baseline.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <CoreFxBaseLinePackageName>Microsoft.Private.PackageBaseline</CoreFxBaseLinePackageName>
     <!-- TODO: Move this back to 1.0.1 once https://github.com/dotnet/corefx/commit/63a868365904c6fbae18a3800c0b908b0f0944af is in -->
```
Comment (ericstj) [inline pkg/baseline/baseline.props]: You can now delete this comment.

Comment (StephenBonikowsky) [inline pkg/baseline/baseline.props]: Will do.

**Comment by dagood** [comment]
Buildtools, CLI, and init-tools changes LGTM.

**Comment by roncain** [comment]
LGTM, subject to CI passing and Eric's and Davis's comments

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by dagood** [comment]
Those ubuntu errors look like they might be legitimate to me:
```
Installing dotnet cli...
Restoring BuildTools version 1.0.26-prerelease-00809-01...
Failed to create prime the NuGet cache. restore failed with: 1
```
I'm going to try to repro, I'll let you know what I see.

**Comment by dagood** [comment]
Nevermind, it got past that this time. :smile:

**Comment by StephenBonikowsky** [comment]
According to @mconnew there is some known issue on some of the virtual machines and choosing "delete slave" on the host fixes it.

**Pull request merged**